lib/silo/commands/file_push.rb

```diff
@@ -40,14 +40,13 @@ def run
 
         source = args[0]
 
-        if args[1]&.match(/^[^:]*:[^:]*$/)
-          silo_name, dest = args[1].split(":").map(&:to_s)
-        else
-          silo_name = Silo.default
-          dest = args[1] || ''
-        end
-
-        dest ||= ''
+        raise InvalidFileNameError, "Target destination '#{args[1]}' contains invalid symbol" unless args[1].nil? || args[1].match?(/^[^:]+(:[^:]+)?$/)
```

Comment:
This still allows the pushing of files with colons in their filenames, etc:
```
[flight@jack-tests-profile (my-cluster) [login1] flight-silo]$ flexec ruby bin/silo set-default jack-silo
Default silo set to: jack-silo
[flight@jack-tests-profile (my-cluster) [login1] flight-silo]$ flexec ruby bin/silo file push :test
Local file '/home/flight/flight-silo/:test' copied to remote '/:test'
```

Trying to pull this file later raises an error:
```
[flight@jack-tests-profile (my-cluster) [login1] flight-silo]$ flexec ruby bin/silo file pull :test
silo: Silo '' not found
```